### PR TITLE
DPC-1011: Unique auto number generator for vendor id

### PR DESCRIPTION
**Pull request by Sun-Mountain**
**Why**

When creating a vendor organization, the ruby app assigns a unique id in place of a provider's NPI for sandbox vendor clients.

**What Changed**

- When an organization is created and `organization_type == 'health_it_vendor`, the rails app auto-generates a vendor id that begins with `"V_"` for easy querying purposes.
- When an organization is updated and `organization_type == 'health_it_vendor` and the `vendor_id` is blank, the rails app auto-generates a vendor id that begins with `V_` for easy querying purposes.
- If a vendor organization is enabled for sandbox, `OrganizationSubmitSerializer` and `FhirResourceBuilder` will use the `vendor_id` as the organization's identifier.

**Choices Made**

- The `vendor_id` will be used in place of an `npi` for an identifier in the sandbox API.
- All `vendor_id`s will start with `"V_"` to make it easy to query vendor users in sandbox API.

**Tickets closed**:

DPC-1011

**Future Work**

N/A

**Checklist**

- [ ] All tests are passing via `make ci-app` (app change) and `make ci-web` (website change)
- [ ] Swagger documentation has been updated
- [ ] FHIR documentation has been updated
- [ ] Any required dpc-ops changes have a PR submitted and mentioned in this ticket
- [ ] Any manual migration steps are documented, scripts written (where applicable), and tested
- [ ] Before merging, any required dpc-ops changes have been approved and merged into master of the dpc-ops repo

### CODE REVIEW

File: dpc-web/app/models/organization.rb [hunk 1]
```diff
@@ -17,6 +17,10 @@ class Organization < ApplicationRecord
   delegate :street, :street_2, :city, :state, :zip, to: :address, allow_nil: true, prefix: true
   accepts_nested_attributes_for :address, reject_if: :all_blank
 
+  has_secure_token :vendor_id
+  # has_secure_token :vendor, if: :vendor <--- does not work (throws an error)
```
Comment (switzersc-usds) [inline dpc-web/app/models/organization.rb]: what's the error it throws?

File: dpc-web/app/models/organization.rb [hunk 2]
```diff
@@ -17,6 +17,10 @@ class Organization < ApplicationRecord
...
+
+  after_create :create_vendor_id
```
Comment (switzersc-usds) [inline dpc-web/app/models/organization.rb]: this shouldn't be necessary with `has_secure_token`

Comment (Sun-Mountain) [reply dpc-web/app/models/organization.rb]: I was using it to run binding.pry and check the last organization created. Didn't know if you wanted to see that.

File: dpc-web/app/models/organization.rb (hunk 1 above)
Comment (Sun-Mountain) [reply dpc-web/app/models/organization.rb]: Screen Shot 2020-02-05 at 2 30 30 PM

**Comment by StewGoin** [comment]
Could someone update the PR to explain what's actually being done in it? I can't tell from reading the template or PR title.

**Comment by Sun-Mountain** [comment]
> Could someone update the PR to explain what's actually being done in it? I can't tell from reading the template or PR title.

Hello! This was a draft of a PR that I'm currently working on to check on a method on assigning users an id. I will update the information, but this PR is not ready for review.

**Comment by Sun-Mountain** [comment]
I'm going to close this and create a cleaner branch for peer review and merge

**Pull request closed without merging**